### DISCUSSION TITLE
Would it be possible to add a slider for one's own microphone sensitivity?

### DISCUSSION THREAD

**Discussion by melolontha-melolontha** [jamulussoftware/jamulus#1497] [UI Related]
Evaluating Jamulus for a choir I listend to quite a few sessions with other people trying out Jamulus. 
One fundamental issue for many beginners is that they don't know where to adjust their microphone's sensitivity. A fader for that purpose in the left part of the Jamulus display would be a great help to many bloody beginners. Of course, well equipped people have a dedicated hardware volume button... 
I think of something which effectively controls the same function which the operating system offers to adapt the input volume, but in some instances (e.g. in Windows) this is not a thing which is always visible like the Jamulus UI when using Jamulus. If real estate in the UI is expensive: cancel the cloud or just make it a turnwheel. But I think space is not an issue, e.g.
![microphone-fader](https://user-images.githubusercontent.com/41631434/114323096-f0bcd580-9b23-11eb-99d6-93c6057f3c1e.png)

**Comment by hoffie** [comment]
> One fundamental issue for many beginners is that they don't know where to adjust their microphone's sensitivity

I agree that this is a huge problem in practice. Sadly, this is highly implementation-specific and I don't think a generic solution for this has been found yet (or would be easy to find).

You can read the previous discussion on that here:
https://github.com/jamulussoftware/jamulus/discussions/1030

**Reply by melolontha-melolontha** [reply]
Yes, and from that one I also read the discussion [https://github.com/jamulussoftware/jamulus/discussions/1030#discussioncomment-554662](https://github.com/jamulussoftware/jamulus/discussions/1030#discussioncomment-554662).
 So there will be something of this type in an upcoming version. The gain control should be some +- dB, i.e. boosting **and** attenuation. Of course attenuation just adds a bit of quantization noise whereas boosting also boosts the result of an already clipped signal (if it was clipped before somewhere). But the larger problem of boosting is to also boost the humming and other electronic background noise. 

BTW: on my Linux computer, there is a program `pavucontrol`. It belongs to one level above ALSA. Jamulus deliberately works with ALSA on Linux. But when I observe `alsamixer`'s settings while moving sliders in `pavucontrol`, I see that they are moved in unison. I guess, `pavucontrol` reads back `alsamixer`'s parameters and sets them. 

For Windows (which I don't actively use any more) googling for "+Windows +10 +control +microphone +gain +API"  led me to https://docs.microsoft.com/en-us/uwp/api/windows.media.devices.audiodevicecontroller?view=winrt-19041. Maybe that's the right hook for it.

**Comment by gilgongo** [comment]
[Input gain](https://github.com/jamulussoftware/jamulus/pull/1222) (but not cut) is tagged for the next release and is I believe now available in the nightly builds.
